### PR TITLE
Clippy fixes v7

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
None 

Describe changes:
- rust: make cargo clippy clean (without the --all-features options, that is without the debug feature)

Follow up after merge of #10156

### CODE REVIEW

**Review by jasonish — APPROVED** [review]
Nice. Some debug log lost, but nothing I'm particularly concerned about.

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on TREX_GENERIC_rule_time.
  

Pipeline 17566

File: rust/src/dcerpc/dcerpc_udp.rs [hunk 1]
```diff
@@ -141,13 +141,7 @@ impl DCERPCUDPState {
...
 
     fn find_incomplete_tx(&mut self, hdr: &DCERPCHdrUdp) -> Option<&mut DCERPCTransaction> {
-        for tx in &mut self.transactions {
-            if tx.seqnum == hdr.seqnum && tx.activityuuid == hdr.activityuuid && ((hdr.pkt_type == DCERPC_TYPE_REQUEST && !tx.req_done) || (hdr.pkt_type == DCERPC_TYPE_RESPONSE && !tx.resp_done)) {
-                SCLogDebug!("found tx id {}, last tx_id {}, {} {}", tx.id, self.tx_id, tx.seqnum, tx.activityuuid[0]);
-                return Some(tx);
-            }
-        }
-        None
+        return self.transactions.iter_mut().find(|tx| tx.seqnum == hdr.seqnum && tx.activityuuid == hdr.activityuuid && ((hdr.pkt_type == DCERPC_TYPE_REQUEST && !tx.req_done) || (hdr.pkt_type == DCERPC_TYPE_RESPONSE && !tx.resp_done)))
```
Comment (victorjulien) [inline rust/src/dcerpc/dcerpc_udp.rs:144]: is this rustfmt clean? Seems like an excessively long line

Comment (catenacyber) [reply rust/src/dcerpc/dcerpc_udp.rs:144]: Indeed, it was not...

Are you ready for https://github.com/OISF/suricata/pull/10088 after this one ?

**Comment by catenacyber** [comment]
Continued in https://github.com/OISF/suricata/pull/10207

**Pull request closed without merging**